pkg/internal/controller/controller.go

```diff
@@ -229,12 +233,15 @@ func (c *Controller) reconcileHandler(obj interface{}) bool {
 	}
 
 	log := c.Log.WithValues("name", req.Name, "namespace", req.Namespace)
+	ctx := logf.IntoContext(context.Background(), log)
```

Comment:
Would `context.TODO()` work a bit better here? It seems like eventually a `WithCancel` context can be plumbed into here from `Start()` that could eventually be used to more cleanly stop in-flight operations rather than just forcefully killing the workers.